### DISCUSSION TITLE
Convert word document to Lexical state

### DISCUSSION THREAD

**Discussion by namoshizun** [facebook/lexical#4448] [Q&A] [answered]
Hiii Lexical team,

I am using Lexical for a knowledgebase system development. Soon I will need to migrate hundreds of Word documents into this system, where the Lexical RTE is used to display & edit them.

Is there some sort of script or Lexical plugin available for converting a docx document into a Lexical state object?

Many thanks!

**Comment by khushal123** [comment]
did you get any solution?

**Comment by ivailop7** [comment, marked as answer]
This is something that is beyond lexical's scope. You will need to find a tool/script that converts docx to standard html and then use generateNodesFromDOM to convert the HTML into Lexical state. [Documentation here](https://lexical.dev/docs/concepts/serialization#html---lexical)
